homeassistant/components/zwave/__init__.py

```diff
@@ -410,28 +409,23 @@ def set_config_parameter(service):
         node = network.nodes[node_id]
         param = service.data.get(const.ATTR_CONFIG_PARAMETER)
         selection = service.data.get(const.ATTR_CONFIG_VALUE)
-        size = service.data.get(const.ATTR_CONFIG_SIZE, 2)
-        i = 0
         for value in (
                 node.get_values(class_id=const.COMMAND_CLASS_CONFIGURATION)
                 .values()):
-            if value.index == param and value.type == const.TYPE_LIST:
-                _LOGGER.debug("Values for parameter %s: %s", param,
-                              value.data_items)
-                i = len(value.data_items) - 1
-        if i == 0:
-            node.set_config_param(param, selection, size)
-        else:
-            if selection > i:
-                _LOGGER.error("Config parameter selection does not exist! "
-                              "Please check zwcfg_[home_id].xml in "
-                              "your homeassistant config directory. "
-                              "Available selections are 0 to %s", i)
-                return
-            node.set_config_param(param, selection, size)
-            _LOGGER.info("Setting config parameter %s on Node %s "
-                         "with selection %s and size=%s", param, node_id,
-                         selection, size)
+            if value.index != param:
```

Comment:
Yes readability:
Take issue #7608 
The provided "value" by the user, does not correspond to the actual value in OZW.
OZW value always corresponds to index within the list of options. But there is no way to retrieve that value from ozw. It always respond with the data or data_as_string.
When retrieving the data_items, it is not sorted in any way, so:
from zwcfg file:
```
				<Value type="list" genre="config" instance="1" index="25" label="Indicate a location for IR code learning and start learning" units="" read_only="false" write_only="true" verify_changes="false" poll_intensity="0" min="0" max="22" vindex="0" size="1">
					<Help>In case none of the code on the code list works for the targeted air conditioner, user can use IR code learning function. See manual at section &quot;IR Code Learning&quot; for a description of the procedure. Value 0-22</Help>
					<Item label="OFF" value="0" />
					<Item label="ON (resume)" value="1" />
					<Item label="cool 19 C" value="2" />
					<Item label="cool 20 C" value="3" />
					<Item label="cool 21 C" value="4" />
					<Item label="cool 22 C" value="5" />
					<Item label="cool 23 C" value="6" />
					<Item label="cool 24 C" value="7" />
					<Item label="cool 25 C" value="8" />
					<Item label="cool 26 C" value="9" />
					<Item label="cool 27 C" value="10" />
					<Item label="cool 28 C" value="11" />
					<Item label="heat 19 C" value="12" />
					<Item label="heat 20 C" value="13" />
					<Item label="heat 21 C" value="14" />
					<Item label="heat 22 C" value="15" />
					<Item label="heat 23 C" value="16" />
					<Item label="heat 24 C" value="17" />
					<Item label="heat 25 C" value="18" />
					<Item label="heat 26 C" value="19" />
					<Item label="heat 27 C" value="20" />
					<Item label="heat 28 C" value="21" />
```
And from the values:
```
25: {
data: "OFF",
data_items: [
"heat 21 C",
"cool 23 C",
"cool 25 C",
"cool 27 C",
"heat 22 C",
"cool 21 C",
"heat 25 C",
"ON (resume)",
"heat 27 C",
"cool 20 C",
"heat 20 C",
"heat 28 C",
"heat 26 C",
"cool 22 C",
"cool 26 C",
"cool 28 C",
"heat 24 C",
"cool 19 C",
"heat 23 C",
"cool 24 C",
"heat 19 C",
"OFF"
],
```